Modules/Microsoft365DSC/DSCResources/MSFT_AADGroup/MSFT_AADGroup.psm1

```diff
@@ -1047,10 +1047,36 @@ function Export-TargetResource
             All         = [switch]$true
             ErrorAction = 'Stop'
         }
-        if ($Filter -like "*endsWith*" -or $Filter -like "*onPremisesSyncEnabled eq null*") {
+
+        # Define the list of attributes
+        $attributesToCheck = @(
+            "description",
+            "displayName",
+            "hasMembersWithLicenseErrors",
+            "mail",
+            "mailNickname",
+            "onPremisesSecurityIdentifier",
+            "onPremisesSyncEnabled",
+            "preferredLanguage"
+        )
+
+        # Initialize a flag to indicate whether any attribute matches the condition
+        $matchConditionFound = $false
+
+        # Check each attribute in the list
+        foreach ($attribute in $attributesToCheck) {
+            if ($Filter -like "*$attribute eq null*") {
+                $matchConditionFound = $true
+                break
+            }
+        }
+
+        # If any attribute matches, add parameters to $ExportParameters
+        if ($matchConditionFound) {
```

Comment:
Now you're missing *endsWith* here, this is not an attribute but a function that you can call on the query which also needs to be checked and if you remove it then it stops working.

Reply:
changed the condition to look as below to take into account endsWith
so it is now

if ($matchConditionFound -or $Filter -like "*endsWith*"))